stacks-signer/src/runloop.rs

```diff
@@ -459,29 +480,109 @@ impl<C: Coordinator> RunLoop<C> {
                 });
             return false;
         };
+
         if block_info.valid.is_none() {
             // We have not yet received validation from the stacks node. Cache the request and wait for validation
             debug!("We have yet to receive validation from the stacks node for a nonce request. Cache the nonce request and wait for block validation...");
-            block_info.nonce_request = Some(request.clone());
+            block_info.nonce_request = Some(nonce_request.clone());
             return false;
         }
-        Self::determine_vote(block_info, request, transactions);
+
+        Self::determine_vote(block_info, nonce_request);
         true
     }
 
+    /// Verify the transactions in a block are as expected
+    fn verify_transactions(&mut self, block: &NakamotoBlock) -> bool {
```

Comment:
I'm not sure if there's test coverage for this?

Reply:
Not sure why i couldn't get the invalid case to not hang in my simulations so have removed that test. I know I check the invalid case in my integration test. But I at least added a unit test for the valid case.